packages/protocol/test/common/multisig.ts

```diff
@@ -0,0 +1,382 @@
+      await assertRevert(multiSig.revokeConfirmation(txId, { from: accounts[2] }))
+    })
+
+    it('should not allow an owner to revoke before confirming', async () => {
+      await assertRevert(multiSig.revokeConfirmation(txId, { from: accounts[1] }))
+    })
+  })
+
+  describe('#addOwner()', () => {
+    it('should allow a new owner to be added via the MultiSig', async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.addOwner(accounts[2]).encodeABI()
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      const txId = txEvent.args.transactionId
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await multiSig.confirmTransaction(txId, { from: accounts[1] })
+      assert.isTrue(await multiSig.isOwner(accounts[2]))
+    })
+
+    it('should not allow an external account to add an owner', async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.addOwner(accounts[2]).encodeABI()
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await assertRevert(
+        multiSig.submitTransaction(multiSig.address, 0, txData, {
+          from: accounts[3],
+        })
+      )
+    })
+
+    it('should not allow adding the null address', async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.addOwner(NULL_ADDRESS).encodeABI()
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      const txId = txEvent.args.transactionId
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await assertRevert(multiSig.confirmTransaction(txId, { from: accounts[1] }))
+    })
+  })
+
+  describe('#removeOwner()', () => {
+    it('should allow an owner to be removed via the MultiSig', async () => {
+      // @ts-ignore
+      const txData = multiSig.contract.methods.removeOwner(accounts[1]).encodeABI()
+
+      const tx = await multiSig.submitTransaction(multiSig.address, 0, txData, {
+        from: accounts[0],
+      })
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txEvent = _.find(tx.logs, {
+        event: 'Confirmation',
+      })
+      // @ts-ignore: TODO(mcortesi): fix typings
+      const txId = txEvent.args.transactionId
+
+      // @ts-ignore: TODO(mcortesi): fix typings
+      await multiSig.confirmTransaction(txId, { from: accounts[1] })
+
+      assert.isFalse(await multiSig.isOwner(accounts[1]))
+      assertEqualBN(await multiSig.required(), 1)
+    })
+
+    it('should not allow an external account to remove an owner', async () => {
```

Comment:
Ditto, should this call `removeOwner` directly?

Reply:
changed